primitives/application-crypto/Cargo.toml

```diff
@@ -18,22 +18,31 @@ targets = ["x86_64-unknown-linux-gnu"]
 sp-core = { version = "7.0.0", default-features = false, path = "../core" }
 codec = { package = "parity-scale-codec", version = "3.0.0", default-features = false, features = ["derive"] }
 scale-info = { version = "2.1.1", default-features = false, features = ["derive"] }
-serde = { version = "1.0.136", optional = true, features = ["derive"] }
+serde = { version = "1.0.136",  default-features = false, optional = true, features = ["derive"] }
 sp-std = { version = "5.0.0", default-features = false, path = "../std" }
 sp-io = { version = "7.0.0", default-features = false, path = "../io" }
 
 [features]
 default = [ "std" ]
 std = [
 	"full_crypto",
+	"serde_full",
 	"sp-core/std",
 	"codec/std",
 	"scale-info/std",
-	"serde",
+	"serde/std",
 	"sp-std/std",
 	"sp-io/std",
 ]
 
+# Serde support without relying on std features.
+serde_full = [
+	"full_crypto",
+	"sp-core/serde_full",
+	"scale-info/serde",
+	"serde/alloc",
```

Comment:
```suggestion
	"dep:serde",
```

Reply:
Thanks for the review! This will make it way less error-prone 👍